dvc/utils/serialize/_common.py

```diff
@@ -42,13 +43,23 @@ def __init__(self, path: "AnyPath", message: str):
         from dvc.utils import relpath
 
         path = relpath(path)
+        self.path = path  # For programmatic consumption, e.g. Studio.
         super().__init__(f"unable to read: '{path}', {message}")
 
 
+class EncodingError(ParseError):
```

Comment:
Isn't this error a _Decoding_ error? :)

Reply:
My thoughts when naming this one were "There is an error with the encoding (noun) of the file".

But as you point out, you can also read this as "An error happened while encoding the file", which isn't the case here.

I'll rename it :)

Reply:
Thinking about it again, I think it's okay. We might be able to reuse this during `dump/modify` without creating yet another `Exception`. And, `encoding` covers both `encode` and `decode` stuff anyway.